packages/perseus/src/widgets/mock-widgets/mock-widget-types.ts

```diff
@@ -0,0 +1,23 @@
+import type {WidgetOptions} from "@khanacademy/perseus-core";
+
+export type MockWidget = WidgetOptions<"mock-widget", MockWidgetOptions>;
+
+export type MockWidgetOptions = {
+    static?: boolean;
+    value: string;
+};
+
+export type PerseusMockWidgetRubric = {
+    value: string;
+};
+
+export type PerseusMockWidgetUserInput = {
+    currentValue: string;
+};
+
+// Extend the widget registries for testing
+declare module "@khanacademy/perseus-core" {
+    export interface PerseusWidgetTypes {
+        "mock-widget": MockWidget;
+    }
+}
```

Comment:
This is the magic sauce that extends the set of `PerseusWidgetTypes` to include the mock widget. Anything that ends up importing this file will "see" the `mock-widget`... meaning tests, but not production code!

Reply:
I don't really understand the code and after looking it up I still don't understand the code. I wonder if would be worth adding a link to something that explains the goal here.

Reply:
I'll expand the comment.  Honestly, I wasn't able to find a really clear description for the case we're using this for.  The bottom line is that when this file is imported  (should be from tests only), the `PerseusWidgetTypes` type from `@khanacademy/perseus-core` is extended to include the `MockWidget`.  Our other types that define the unions of widget options types, etc. are all based on `PerseusWidgetTypes`. So once we extend `PerseusWidgetTypes` with this mock widget type, all the other widget option unions know about the mock widget also. This means that our tests and test data understand that `mock-widget 1` refers to a mock widget, but if you were to try it in webapp, it wouldn't know about mock widgets (rightfully).